src/listen/heosResponse.ts

```diff
@@ -0,0 +1,40 @@
+import { HeosResponse, HeosResponseMessage } from '../types'
+import {
+	FIRST_LEVEL_HEOS_RESPONSE_MESSAGE_SPLITTER,
+	SECOND_LEVEL_HEOS_RESPONSE_MESSAGE_SPLITTER
+} from '../utils/constants'
+
+export function parseHeosMessageString(
+	messageString: string | undefined
+): HeosResponseMessage | string | void {
```

Comment:
Could it be easier to always return a `HeosResponseMessage`? If row 8 looked like `messageString: string = ''` It would always default to a string and therefore the if case on row 10 might not be necessary.

Reply:
True but if the response from HEOS doesn't have a message attribute why should this function return one.